aviatrix/resource_aviatrix_kubernetes_cluster.go

```diff
@@ -0,0 +1,291 @@
+				RequiredWith: []string{"account_id", "name", "vpc_id", "region", "version", "platform", "is_publicly_accessible", "network_mode"},
+			},
+			"account_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				Description:  "Account ID owning the cluster",
+				RequiredWith: []string{"account_name", "name", "vpc_id", "region", "version", "platform", "is_publicly_accessible", "network_mode"},
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Display name of the cluster",
+				RequiredWith: []string{"account_id", "account_name", "vpc_id", "region", "version", "platform", "is_publicly_accessible", "network_mode"},
+			},
+			"vpc_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Id of the VPC where the cluster is deployed",
+				RequiredWith: []string{"account_id", "account_name", "name", "region", "version", "platform", "is_publicly_accessible", "network_mode"},
+			},
+			"region": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Region where the cluster is deployed.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "version", "platform", "is_publicly_accessible", "network_mode"},
+			},
+			"version": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Version of the Kubernetes cluster.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "platform", "is_publicly_accessible", "network_mode"},
+			},
+			"platform": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Platform of the Kubernetes cluster, e.g. kops, kubeadm or any other free string.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "version", "is_publicly_accessible", "network_mode"},
+			},
+			"is_publicly_accessible": {
+				Type:         schema.TypeBool,
+				Optional:     true,
+				Description:  "Whether the API server is publicly accessible outside the virtual network.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "version", "platform", "network_mode"},
+			},
+			"network_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Network mode of the cluster. Possible values are FLAT, OVERLAY.",
+				ValidateFunc: validation.StringInSlice([]string{"FLAT", "OVERLAY"}, false),
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "version", "platform", "is_publicly_accessible"},
+			},
+			"project": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Project name if the cluster is deployed in GCP.",
+			},
+			"compartment": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Compartment id if the cluster is deployed in OCI.",
+			},
+			"tags": {
```

Comment:
Are there 'tags' always applicable? Even with a cluster that we scrape from CSP that already has its own tags?

Reply:
They should only be applicable to custom clusters, these are not intended for clusters discovered from the CSP.
I'll try to figure out how to validate it accordingly.

Reply:
It should be clearer no with the tags being in the cluster_resource.